src/nil.cr

```diff
@@ -106,6 +106,19 @@ struct Nil
     raise NilAssertionError.new
   end
 
+  # Returns `self`
+  # Added to make the code below compile.
```

Comment:
Please rephrase this so that it actually makes sense for someone who hasn't seen this PR.

For example: ``This method enables to call the `presence` method (see `String#presence`) on a union with `Nil`. The idea is to return `nil` when the value is `nil` or empty.``